### ISSUE TITLE
Regression in 1.1.8: _is_resource_annotated throws TypeError: issubclass() arg 1 must be a class

### ISSUE THREAD

**Issue by mikeatlas** [dagster-io/dagster#11541]
### Dagster version

1.1.8

### What's the issue?

Our job definitions test suite no longer passes with 1.1.8 - previously it worked with 1.1.7.

The code change in question appears to originate from https://github.com/dagster-io/dagster/pull/11322 - [Allow specifying resources in Op/Asset params list #11322](https://github.com/dagster-io/dagster/pull/11322)

Our failing tests stack trace looks like this:

```
___ ERROR collecting tests/my_test.py ___
 [ ... snip ... ]
    def do_something(context, messages: list[dict[str, Any]]):
dagster/_core/definitions/decorators/op_decorator.py:86: in __call__
    arg_resource_keys = {arg.name for arg in compute_fn.get_resource_args()}
dagster/_core/definitions/decorators/solid_decorator.py:65: in get_resource_args
    return get_resource_args(self.decorated_fn)
dagster/_core/definitions/resource_output.py:12: in get_resource_args
    return [param for param in get_function_params(fn) if _is_resource_annotated(param)]
dagster/_core/definitions/resource_output.py:12: in <listcomp>
    return [param for param in get_function_params(fn) if _is_resource_annotated(param)]
dagster/_core/definitions/resource_output.py:17: in _is_resource_annotated
    isinstance(param.annotation, type) and issubclass(param.annotation, ResourceDefinition)
python3.10/abc.py:123: in __subclasscheck__
    return _abc_subclasscheck(cls, subclass)
E   TypeError: issubclass() arg 1 must be a class
```

The op in question that this fails on is relatively benign looking:
```python
@op(
    required_resource_keys={"abc"},
    ins={"messages": In()},
)
def do_something(context, messages: list[dict[str, Any]]):
```

### What did you expect to happen?

Our tests do not even exercise our `op` function with this issue as it appears the test runner fails to load each unit test.

### How to reproduce?

Create a python test that invokes the op as normal function, and try to run with pytest.

```python
from my_job_defs import do_something

def test_do_something():
    assert do_something(null, []) == None
```
Run with pytest (note that the test itself does not even run the assertion line of code - it fails on module import).

### Deployment type

Other

### Deployment details

This can be surfaced without a deployment.

### Additional information

I am on the Dagster Slack and you can ping me with questions or ask here.

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by mikeatlas** [comment]
tagging @benpankow for awareness

**Comment by ynotzort** [comment]
have the same issue:

this fails:
```python
@asset()
def asset_a() -> dict[str, int]:
    return {"a": 123}

@asset()
def asset_b(asset_a: dict[str, int]) -> int:
    return 123
```

but this works:
```python
from typing import Dict
@asset()
def asset_a() -> dict[str, int]:
    return {"a": 123}

@asset()
def asset_b(asset_a: Dict[str, int]) -> int:
    return 123
```

**Comment by mikeatlas** [comment]
@ynotzort there's a fix that was just merged, I'm guessing a release will follow soon enough. For us, we can hold off and skip 1.1.8 update.

**Comment by benpankow** [comment]
We're shipping out a release right now to fix this issue, thank you for your patience!

**Comment by mikeatlas** [comment]
Woo! Patience... ha! Your turnaround on this is very fast. Thank you.

**Comment by mikeatlas** [comment]
Confirmed fixed in 1.1.9 @benpankow 🎉
